core/dbt/events/base_types.py

```diff
@@ -74,8 +74,13 @@ def __init__(self, *args, **kwargs):
             from dbt.events.types import Note
             from dbt.events.functions import fire_event
 
-            fire_event(Note(msg=f"[{class_name}]: Unable to parse dict {kwargs}"))
-            self.pb_msg = msg_cls()
+            error_msg = f"[{class_name}]: Unable to parse dict {kwargs}"
+            # If we're testing throw an error so that we notice failures
+            if "pytest" in sys.modules:
+                raise Exception(error_msg)
+            else:
+                fire_event(Note(msg=error_msg))
+                self.pb_msg = msg_cls()
```

Comment:
ah! this is good!

Reply:
does this also mean that "fire a warning instead of raising an exception on event serialization errors" was effectively resolved by the proto changes?
- https://github.com/dbt-labs/dbt-core/issues/7113

Reply:
Well, probably, but mainly because the google protobuf messages are more strongly typed, so that errors are caught here instead of on serialization. As a belt-and-suspenders thing it's probably not a bad idea to catch on serialization too (though I haven't actually seen any errors there).